atf: Bring all boards to the same ATF SHA

bryanodonoghue opened the pull request:
This commit brings all boards to the same ATF SHA, based off of ATF 2.1
with our out-of-tree patches for Pico and i.MX8MM-EVK stacked on top.

- Built and tested on i.MX8, currently building and testing for RPI3, Pico and WaRP7

bryanodonoghue commented:
http://jenkins.mbed-linux.arm.com/job/bod-mbl-master/236

diego-sueiro commented on meta-mbl-bsp-common/recipes-bsp/atf/atf.inc:
```diff
@@ -14,10 +14,13 @@ SRC_URI = " \
     git://github.com/ARMmbed/mbedtls.git;protocol=https;branch=development;name=mbedtls;destsuffix=mbedtls \
 "
 
+# MBL_ATF_VERSION should be updated to match version pointed to by SRCREV
```
2 observations:

1. The atf build directory should be under the machine's WORKDIR, and for Rpi it is under the core architecture (e.g. `cortexa7t2hf-neon-vfpv4-oe-linux-gnueabi`);
2. We should set the atf package version properly.

My suggestion is to add the following lines to this file:

```
PACKAGE_ARCH = "${MACHINE_ARCH}"
PV = "${MBL_ATF_VERSION}+git${SRCPV}"
```

bryanodonoghue replied:
The WORKDIR for rpi3 is not affected by my patch but, I can look at it since you've raised it.
The package version I think is an easy change.

bryanodonoghue commented:
I've
* added the PV
* Tested on imx8
* Tested on rpi3
* Updated the jenkins build with LAVA parameters.

http://jenkins.mbed-linux.arm.com/job/bod-mbl-master/238

bryanodonoghue commented:
OK once more : http://jenkins.mbed-linux.arm.com/job/bod-mbl-master/239

diego-sueiro commented:
> OK once more : http://jenkins.mbed-linux.arm.com/job/bod-mbl-master/239

The build for pico7 got stuck.

Started another one for this machine: http://jenkins.mbed-linux.arm.com/job/ds-warrior-dev/49/

diego-sueiro commented:
> > OK once more : http://jenkins.mbed-linux.arm.com/job/bod-mbl-master/239
> 
> The build for pico7 got stuck.
> 
> Started another one for this machine: http://jenkins.mbed-linux.arm.com/job/ds-warrior-dev/49/

All green.
